pkg/subscription/engine.go

```diff
@@ -177,6 +178,26 @@ func (e *ExecutorEngine) executeSubscription(buf *graphql.EngineResultWriter, id
 	}
 }
 
+// executeWithBackOff runs the executor wrapped in an exponential backOff algorithm of t=b^c
+func (e *ExecutorEngine) executeWithBackOff(id string, executor Executor, buf *graphql.EngineResultWriter, eventHandler EventHandler) error {
+	nextRetry := time.Second
+	var err error
+	for {
+		err = executor.Execute(buf)
+		if err == nil || !strings.Contains(err.Error(), "failed to WebSocket dial") {
```

Comment:
maybe we could check/compare against the error type (`errors.Is` or `errors.As`) instead of matching a part of the string.